android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -31,10 +33,33 @@ public class CameraCaptureController extends AbstractVideoCaptureController {
      */
     private final CameraEventsHandler cameraEventsHandler = new CameraEventsHandler();
 
+    @Override
+    public void startCapture() {
+        super.startCapture();
+        this.isCapturing = true;
+        // Checking if we need to switch the camera
+        if(facingModeWhenStoppedCapturing != null && facingModeWhenStoppedCapturing != facingMode()) {
+            CameraCaptureController.this.switchCamera(new SwitchCameraHandler() {
```

Comment:
In the case that the user tries to switch while the device is off, will we end up firing `onSwitchCameraDone()` twice? (Once on line 81 and then again when we _actually_ do the switch)? And if so, what's the effect of that? I think it's used to resolve the promise when the user calls `callObject.cycleCamera()` or `callObject.setCamera()`.

Reply:
No, It is only going to fire to the user on line `81`.

Here on this line, we are just passing this handler to log, so we can be aware that we have restored the camera from a wrong state, but this event is not going to the user. As you can see in this case, we are creating the `SwitchCameraHandler`, we are not receiving It from the user request.

Reply:
Ah right, `onSwitchCameraDone` would have to be implemented in this handler for it to be called again! Cool cool.